x/evm/keeper/state_transition.go

```diff
@@ -347,8 +350,12 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context,
 		// NOTE: this only adds active precompiles to the EVM.
 		// This means that evm.Precompile(addr) will return false for inactive precompiles
 		// even though this is actually a reserved address.
-		precompileMap := k.Precompiles(activePrecompiles...)
-		evm.WithPrecompiles(precompileMap, activePrecompiles)
+		activePrempilesMap := k.GetStaticPrecompilesInstances(activeStaticPrecompiles...)
+		dynamicPrecompileMap := k.GetDynamicPrecompileInstance(ctx, dynamicPrecompiles...)
```

Comment:
we can prevent this instantiation if the contract address is not interacting with the precompile (eg check the to address, any internal calls or skip if the tx is not CALL).

This will make the txs way faster

Reply:
We can't do it! because we dont know about internal transactions (contracts calling the precompile contract)

Reply:
Good thing is that instantiation is not expensive anymore since abi loads only on init

Reply:
Re reading the examples you provided - Yes we can skip for transfers and deployments but not for contract calls